src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs

```diff
@@ -81,7 +81,13 @@ public static void ExtractToFile(this ZipArchiveEntry source, string destination
                 ExtractExternalAttributes(fs, source);
             }
 
-            File.SetLastWriteTime(destinationFileName, source.LastWriteTime.DateTime);
+            try
+            {
+                File.SetLastWriteTime(destinationFileName, source.LastWriteTime.DateTime);
+            }
+            catch (UnauthorizedAccessException) when (OperatingSystem.IsAndroid()) // not every Android supports that (#35374)
```

Comment:
Why special-case Android?  If an UnauthorizedAccessException gets thrown on iOS, are we going to add to the when?  What about on Linux? On Windows?  I'm wondering if we really need the when at all.

Reply:
I agree, and we want to reduce usage of exception filters since they are problematic on certain platforms.